test/object-store/sync/app.cpp

```diff
@@ -3248,6 +3248,262 @@ TEST_CASE("app: sync integration", "[sync][pbs][app][baas]") {
     }
 }
 
+TEST_CASE("app: network transport handles redirection", "[sync][app][baas]") {
+    auto logger = util::Logger::get_default_logger();
+    auto app_session = get_runtime_app_session();
+
+    // Skip this test if not using the redirect server
+    if (!get_redirector())
```

Comment:
Why wouldn't we just explicitly create a redirecting server for this test regardless of whether it's enabled globally?